### DISCUSSION TITLE
How to run the custom provider outside of the host?

### DISCUSSION THREAD

**Discussion by unitecoder** [wasmCloud/wasmCloud#3811] [Q&A] [answered]
Hello, everyone. I'm new to wasmCloud and I really like this project. It's awesome and I want to use it in my project. 

I have developed a custom Go provider for a demo. However, it seems I can only run my provider on a wasmCloud host. This makes it hard for me to debug because I can't run the provider in my IDE and connect to the wasmCloud host for debugging.

I tried to navigate to the page https://wasmcloud.com/docs/developer/troubleshooting/providers to find troubleshooting tips for providers, but the page is still under construction.

So, I am looking for a solution to run my custom provider outside the host. Is there a way to do this?

Besides, I also want my provider to run independently not only for the debugging. This will make deploying the provider more flexible.

**Comment by thomastaylor312** [comment, marked as answer]
We actually do have an example of this, but it isn't well documented as you pointed out! https://github.com/wasmCloud/wasmCloud/blob/d0ebbd0b1aa4a9e799084d546b075e0a83f87bdc/examples/golang/providers/custom-template/run.sh is the script that runs it. Instructions for calling it can be found in the README: https://github.com/wasmCloud/wasmCloud/blob/d0ebbd0b1aa4a9e799084d546b075e0a83f87bdc/examples/golang/providers/custom-template/README.md

I've opened this issue to track adding more documentation: https://github.com/wasmCloud/wasmcloud.com/issues/773
